model/convert/service_event.go

```diff
@@ -943,57 +968,39 @@ func convertClusterQCVotes(cdcClusterQCs []cadence.Value) (
 // convertDKGKeys converts hex-encoded DKG public keys as received by the DKG
 // smart contract into crypto.PublicKey representations suitable for inclusion
 // in the protocol state.
-func convertDKGKeys(cdcDKGKeys []cadence.Value) (
-	groupKey crypto.PublicKey,
-	participantKeys []crypto.PublicKey,
+func convertDKGKeys(cdcDKGKeys ...cadence.Value) (
+	convertedKeys []crypto.PublicKey,
 	err error,
 ) {
 
 	hexDKGKeys := make([]string, 0, len(cdcDKGKeys))
 	for _, value := range cdcDKGKeys {
 		keyHex, ok := value.(cadence.String)
 		if !ok {
-			return nil, nil, invalidCadenceTypeError("dkgKey", value, cadence.String(""))
+			return nil, invalidCadenceTypeError("dkgKey", value, cadence.String(""))
 		}
 		hexDKGKeys = append(hexDKGKeys, string(keyHex))
 	}
```

Comment:
I feel we are just needlessly allocating the slice `hexDKGKeys` for interim results here and iterating over the keys twice. 

Could the following work:
```golang
// convertDKGKeys converts hex-encoded DKG public keys as received by the DKG
// smart contract into crypto.PublicKey representations suitable for inclusion
// in the protocol state.
func convertDKGKeys(cdcDKGKeys ...cadence.Value) ([]crypto.PublicKey, error) {
	convertedKeys := make([]crypto.PublicKey, 0, len(cdcDKGKeys))
	for _, value := range cdcDKGKeys {
		pubKey, err := convertDKGKey(value)
		if err != nil {
			return nil, fmt.Errorf("could not decode dkg public key: %w", err)
		}
		convertedKeys = append(convertedKeys, pubKey)
	}
	return convertedKeys, nil
}

// convertDKGKey converts a single hex-encoded DKG public key as received by the DKG
// smart contract into crypto.PublicKey representations suitable for inclusion
// in the protocol state.
func convertDKGKey(cdcDKGKeys cadence.Value) (crypto.PublicKey, error) {
	// extract string representation from Cadence Value
	keyHex, ok := cdcDKGKeys.(cadence.String)
	if !ok {
		return nil, invalidCadenceTypeError("dkgKey", cdcDKGKeys, cadence.String(""))
	}

	// decode individual public keys
	pubKeyBytes, err := hex.DecodeString(string(keyHex))
	if err != nil {
		return nil, fmt.Errorf("could not decode individual public key into bytes: %w", err)
	}
	pubKey, err := crypto.DecodePublicKey(crypto.BLSBLS12381, pubKeyBytes)
	if err != nil {
		return nil, fmt.Errorf("could not decode dkg public key: %w", err)
	}
	return pubKey, nil
}
```

Reply:
[`0e9e8d8` (#6494)](https://github.com/onflow/flow-go/pull/6494/commits/0e9e8d88fef64bcf858ab7a609be2eeb07cc601b)

Very good, thanks.